rpc/errors.go

```diff
@@ -63,46 +66,28 @@ func tryUnwrapToRPCErr(err error, rpcErrors ...*RPCError) error {
 	return Err(InternalError, err)
 }
 
-// isErrUnexpectedError checks if the error is an unexpected error.
+// isErrorWithData checks if the error is the type of error that might contain information in the data field.
+// In the case it is, it adds this information to the returned error.
 //
 // Parameters:
-// - err: The error to be checked
+// - nodeErr: The error to be checked
 // Returns:
-// - *RPCError: a pointer to an RPCError object
-// - bool: a boolean value indicating if the error is an unexpected error
-func isErrUnexpectedError(err error) (*RPCError, bool) {
-	var nodeErr *RPCError
-	if json.Unmarshal([]byte(err.Error()), nodeErr) != nil {
-		return nil, false
-	}
-
+// - *RPCError: a pointer to the RPCError resulting object
+// - bool: a boolean value indicating if the error is of the same error type as `rpcErr`
+func isErrorWithData(nodeErr *RPCError) (*RPCError, bool) {
```

Comment:
nit: I don't think we need to return the `bool`. We can just return `*RPCError` and check if it is `nil`.